lib/internal/streams/async_iterator.js

```diff
@@ -155,7 +155,7 @@ const createReadableStreamAsyncIterator = (stream) => {
   });
   iterator[kLastPromise] = null;
 
-  finished(stream, (err) => {
+  finished(stream, { writable: false }, (err) => {
```

Comment:
this PR also contains a minor adjustment for async_iterator. Since it is only for readable streams, it is therefore also only for the readable side of duplex/transform streams. Certain cases allow for the readable side of a stream to close without the writable side ending. The `finished` function by default only calls the done callback when the writable side of a stream is done. This leads to certain edge cases in the Transform.by implementation which can make a `for await` loop hang on an unresolved final promise. Setting `writable` to `false` tells `finished` to call the done callback when the readable side of a duplex/transform stream closes.

Reply:
cc @mcollina

Reply:
Can you move this into his own commit? it might be worthwhile having it in its own PR.

Reply:
If I do it will break a test in this PR - what’s the convention on dependent PR’s?

Reply:
when we rebase it'll go away or conflict.

Reply:
https://github.com/nodejs/node/pull/28566